src/cdk-lib/bedrock/guardrails/guardrail-filters.ts

```diff
@@ -0,0 +1,428 @@
+   * denounce, or dehumanize a person or group on the basis of an identity (such as race,
+   * ethnicity, gender, religion, sexual orientation, ability, and national origin).
+   */
+  HATE = 'HATE',
+  /**
+   * Describes input prompts and model responses that includes demeaning, humiliating,
+   * mocking, insulting, or belittling language. This type of language is also labeled
+   * as bullying.
+   */
+  INSULTS = 'INSULTS',
+  /**
+   * Describes input prompts and model responses that seeks or provides information
+   * about engaging in misconduct activity, or harming, defrauding, or taking advantage
+   * of a person, group or institution.
+   */
+  MISCONDUCT = 'MISCONDUCT',
+  /**
+   * Enable to detect and block user inputs attempting to override system instructions.
+   * To avoid misclassifying system prompts as a prompt attack and ensure that the filters
+   * are selectively applied to user inputs, use input tagging.
+   */
+  PROMPT_ATTACK = 'PROMPT_ATTACK',
+}
+
+/**
+ * Interface to declare a content filter.
+ */
+export interface ContentFilter {
+  /**
+   * The type of harmful category that the content filter is applied to
+   */
+  readonly type: ContentFilterType;
+  /**
+   * The strength of the content filter to apply to prompts / user input.
+   */
+  readonly inputStrength: ContentFilterStrength;
+  /**
+   * The strength of the content filter to apply to model responses.
+   */
+  readonly outputStrength: ContentFilterStrength;
+}
+
+/******************************************************************************
+ *                              TOPIC FILTERS
+ *****************************************************************************/
+/**
+ * Defines a topic to deny.
+ */
+export interface DeniedTopic {
+  /**
+   * The name of the topic to deny.
+   */
+  readonly name: string;
+  /**
+   * Provide a clear definition to detect and block user inputs and FM responses
+   * that fall into this topic. Avoid starting with "don't".
+   * @example `Investment advice refers to inquiries, guidance, or recommendations
+   * regarding the management or allocation of funds or assets with the goal of
+   * generating returns or achieving specific financial objectives.`
+   */
+  readonly definition: string;
+  /**
+   * Representative phrases that refer to the topic. These phrases can represent
+   * a user input or a model response. Add up to 5 phrases, up to 100 characters
+   * each.
+   * @example "Where should I invest my money?"
+   */
+  readonly examples?: string[];
+}
+
+/******************************************************************************
+ *                               WORD FILTERS
+ *****************************************************************************/
+/**
+ * The managed word type filter available for guardrails.
+ */
+export enum ManagedWordFilterType {
+  PROFANITY = 'PROFANITY',
+}
+
+export type WordFilter = string[] | ManagedWordFilterType[];
```

Comment:
Need to test this in other languages, but the JSII reference mentions that type aliasing won't generate an error at compilation time, but the behavior is different for TypeScript compared to other language targets. See https://aws.github.io/jsii/user-guides/lib-author/typescript-restrictions/ , section "Type aliases"